src/bootstrap/src/utils/execution_context.rs

```diff
@@ -89,12 +123,28 @@ impl ExecutionContext {
         stderr: OutputMode,
     ) -> DeferredCommand<'a> {
         command.mark_as_executed();
```

Comment:
I wonder if we should only mark_as_executed in finish, because if someone forgets to finish a started command, that's also not a good thing. At least it would be nice to mark DeferredCommand with #[must_use].